neuralprophet/components/seasonality/fourier.py

```diff
@@ -0,0 +1,154 @@
+from abc import abstractmethod
+
+import torch
+import torch.nn as nn
+
+from neuralprophet import utils
+from neuralprophet.components.seasonality import Seasonality
+from neuralprophet.utils_torch import init_parameter
+
+
+class FourierSeasonality(Seasonality):
+    def __init__(self, config, id_list, quantiles, num_seasonalities_modelled, n_forecasts, device):
+        super().__init__(
+            config=config,
+            n_forecasts=n_forecasts,
+            num_seasonalities_modelled=num_seasonalities_modelled,
+            quantiles=quantiles,
+            id_list=id_list,
+            device=device,
+        )
+        self.season_dims = utils.config_seasonality_to_model_dims(self.config_seasonality)
+        if self.season_dims is not None:
+            # Seasonality parameters for global or local modelling
+            self.season_params = nn.ParameterDict(
+                {
+                    # dimensions - [no. of quantiles, num_seasonalities_modelled, no. of fourier terms for each seasonality]
+                    name: init_parameter(dims=[len(self.quantiles)] + [self.num_seasonalities_modelled] + [dim])
+                    for name, dim in self.season_dims.items()
+                }
+            )
```

Comment:
In a future PR, we may want to have a seasonality object per period instantiated - thus a class should not need know the `num_seasonalities_modelled`.